src/modules/BootSupportCheck.rb

```diff
@@ -103,11 +101,11 @@ def correct_loader_type(lt)
     def check_gpt_reserved_partition
       return true unless stage1.mbr?
 
-      disk = staging.disks.with(name: BootStorage.mbr_disk).first
-      boot_device = staging.partitions.with(name: BootStorage.BootPartitionDevice).first
+      disk = BootStorage.mbr_disk
+      boot_device = BootStorage.boot_partition
       return true unless disk.gpt?
-      return true if !boot_device.has_filesystem || boot_device.filesystem.type != ::Storage::FsType_BTRFS
-      return true if disk.partitions.any? { |p| p.partition_id == ::Storage::ID_BIOS_BOOT }
+      return true if boot_device.filesystem_type != ::Y2Storage::Filesystems::Type::BTRFS
```

Comment:
Another possibility (just for completeness, not implying it's better than the current one)
```ruby
return true if boot_device.filesystem.nil? || !boot_device.filesystem.type.is?(:btrfs)
```